### PR TITLE
implement `pg_catalog.pg_type` table

### PR DESCRIPTION
**Pull request by universalmind303**
closes https://github.com/GlareDB/glaredb/issues/747
closes https://github.com/GlareDB/glaredb/issues/2390

Note, I don't think this quite yet closes the other [connector issues](https://github.com/GlareDB/glaredb/issues?q=is%3Aissue+is%3Aopen+pg_type). Many of them rely on transactions to work properly (such as https://github.com/GlareDB/glaredb/issues/1450)

### CODE REVIEW

File: crates/glaredb/src/commands.rs [hunk 1]
```diff
@@ -281,7 +281,22 @@ impl RunCommand for MetastoreArgs {
...
+    //
+    // Note that Sean observed the stack size only reaching ~300KB when
+    // running in release mode, and so we don't need to bump this
+    // everywhere. However there's definitely improvements to stack
+    // usage that we can make.
+    // see <https://github.com/GlareDB/glaredb/issues/2390>
+    #[cfg(not(release))]
+    builder.thread_stack_size(4 * 1024 * 1024);
+
+    let runtime = builder
```
Comment (universalmind303) [inline crates/glaredb/src/commands.rs:299]: the sql queries from some of the connectors was causing stack overflows in dev due to referencing many views.

**Review by tychoish — COMMENTED** [review]
is there any kind of test we should write? 

is the nested view stack problem the kind of thing that we should raise with datafusion? It seems like we don't really have enough control or tools to push the stack size back down?

**Comment by universalmind303** [comment]
> is there any kind of test we should write?
> 
> is the nested view stack problem the kind of thing that we should raise with datafusion? It seems like we don't really have enough control or tools to push the stack size back down?

release already uses different optimizations & removes all debug symbols, so this isn't really an issue outside of development. Since this is something we've already done with our SLT for the exact same reason, it feels reasonable to introduce it here as well. 

its really only an issue in develop, and it is a somewhat known one. There are a few issues already [open](https://github.com/apache/arrow-datafusion/issues?q=is%3Aissue+stack+overflow+is%3Aopen) in datafusion wrt SO during planning.

**Review by scsmithr — COMMENTED** [review]
This looks good. Let's add an SLT to assert we can query this without erroring.

**Review by vrongmeal — APPROVED** [review]

**Pull request merged**